Sources/ConsoleKit/Utilities/ConsoleLogger.swift

```diff
@@ -1,5 +1,96 @@
 import Logging
 
+/// A `LoggerFragment` which implements the default logger message format.
+public func defaultLoggerFragment() -> some LoggerFragment {
+	LabelFragment().maxLevel(.trace)
+		.and(LevelFragment().separated(" ").and(MessageFragment().separated(" ")))
+		.and(MetadataFragment().separated(" "))
+		.and(FileFragment().separated(" ").maxLevel(.debug))
+}
+
+/// Outputs logs to a `Console` via a `LoggerFragment` pipeline.
+public struct ConsoleFragmentLogger<T: LoggerFragment>: LogHandler {
```

Comment:
We need to ensure this is Sendable since LogHandlers are now Sendable

Reply:
Does this imply `LoggerFragment` should be `Sendable`? It looks like doing that would require making `ConsoleText` `Sendable` as well.

Reply:
Actually even adding the explicit `Sendable` to just the log handlers produces a warning about `Console` not being `Sendable`

Reply:
Yeah and so starts the rabbit hole thread pulling of Sendable. We either need to wrap stuff in `NIOLoopBound` if we can guarantee that we're going to be on an event loop or just make it Sendable

Reply:
Looking at it, ConsoleText (and the types it pulls in) can all be marked as Sendable without any issue. You can slap a Sendable on Console and see what breaks

Reply:
Terminal will be the issue because of the user info dictionary that's `Any`. I'm happy to make that `Sendable` instead as that just produces warnings for types that aren't Sendable

Reply:
I went ahead and pushed some changes. I'm still getting a warning on `userInfo` since it's mutable. Is adding a dependency on NIO worth it for getting the locked box? Should I use NSLock? 

I also had to modify `VeryUnsafeMutableSigAtomicBufferPointer`. The data structure itself is actually immutable but it was using a subscript to mutate the underlying pointer which meant the global variable had to be `var` which strict concurrency checking didn't like. I just made manual methods for getting and setting. That hurts the ergonomics pretty badly, but it's 1. an implementation detail and 2. not used very much. A closure accepting an inout param would have been better, but I wasn't confident that wouldn't run afoul of the strongly worded warnings on the type so I just settled for the simplest option for now.

Reply:
@0xTim do you have thoughts on what to do for the `userInfo` fields? Using a lock with separate getters and setters sounds messy but I'm not sure if there's a better solution with this API.

Reply:
Ah because it's on a class. Yeah wrap it in `NIOLockedValueBox` in a new property called `_userInfo` and then make `userInfo` a computed property that accessed the underlying value via the lock to maintain the public API

Reply:
@0xTim ConsoleKit doesn't currently depend on NIO, so we'd have to add it as a dependency to use NIOLockedValueBox.

Reply:
That might be a road we want to go down anyway (adding the NIO dependency), since ideally ConsoleKit's API should support Concurrency (and to a certain extent, pre-Concurrency ELF mechanisms).

Reply:
It would also help ease the planned (eventual) transition to fully deprecating the existing `Command` definitions in favor of first-class support for `swift-argument-parser`.

Reply:
Yeah I have no problem adding it as a dependency. I can't imagine there are many use cases where it's not pulled in by something else

Reply:
(That being said, in an ideal world the next major release won't need it and have a full Concurrency support)

Reply:
Ahh `AnyHashable` isn't `Sendable`. Is there a way to work around that other than creating a new `@unchecked Sendable` wrapper?

Reply:
Can we not use `Hashable & Sendable`? Or do we need to type-erase it because old swift and multiple types?

Reply:
Even in Swift 5.7 you can't just write `Dictionary<some Hashable & Sendable, any Sendable>`, and I don't believe `any Hashable` works as a key either (though arguably the underlying category theory says it should). But you can write `any (AnyHashable & Sendable)`.

Reply:
Ah, right, sorry, my mistake 🤦‍♀️

Reply:
I think our only option that retains public API compatibility and avoids Concurrency warnings is to trick the compiler by redeclaring `AnyHashable`. It'll be a blighted mess, but it might work:
```swift
/// Ref: https://github.com/apple/swift/blob/main/stdlib/public/core/AnyHashable.swift
@_spi(AnyHashableSendabilityWorkaround)
public struct AnyHashable: Hashable, CustomStringConvertible, CustomDebugStringConvertible, CustomReflectable, _HasCustomAnyHashableRepresentation, @unchecked Sendable {
    var base: Swift.AnyHashable
    public init<H: Hashable & Sendable>(_ base: H) { self.base = .init(base) }
    public var base: Any { self.base.base }
    public static func == (lhs: ConsoleKit.AnyHashable, rhs: ConsoleKit.AnyHashable) -> Bool { lhs.base == rhs.base }
    public var hashValue: Int { self.base.hashValue }
    public func hash(into hasher: inout Hasher) { self.base.hash(into: &hasher) }
    public func _rawHashValue(seed: Int) -> Int { self.base._rawHashValue(seed: seed) }
    public var description: String { self.base.description }
    public var debugDescription: String { self.base.debugDescription }
    public var customMirror: Mirror { self.base.customMirror }
    @_alwaysEmitIntoClient public __consuming func _toCustomAnyHashable() -> AnyHashable? { self.base._toCustomAnyHashable() }
}

Reply:
Technically this is still breaking since non-`Sendable` inputs will no longer be accepted... we'll probably have to remove that requirement and thus _deliberately_ and completely bypass the entire `Sendable` checking machinery, but I don't see much else in the way of options.

Reply:
It's worth checking this. Will they not be accepted or will they produce a warning? Because I'd assume that's the case

Reply:
You can't use an @_spi type in a public interface can you? Even if we did that it would still break client code because they wouldn't be the same `AnyHashable` type I think?

Reply:
No, they'll be rejected; it's an explicit protocol conformance constraint, not an implied sendability violation.

Reply:
> You can't use an @_spi type in a public interface can you? Even if we did that it would still break client code because they wouldn't be the same `AnyHashable` type I think?

Damn, you're right 🤦‍♀️ And we can't hide it by giving it a leading underscore either...

Reply:
Actually, we don't have to hide it, come to think of it. The worst we'd ever have to do, assuming the stdlib version gains Sendabiltiy at some point, is change it to `public typealias AnyHashable = Swift.AnyHashable` - since the public API is being mimicked method for method, as long as it can still be referred to by module-qualified name, it doesn't matter which version people get (aside from the sendable question)

Reply:
Well the names will conflict if ours is public. And if an API was previously passing a Swift.AnyHashable as a key directly that would break since that would no longer be a compatible type.

This entire API is extremely not `Sendable` friendly 😅

Reply:
Tim and I are currently discussing the best approach, bear with us 😅

Reply:
@semicoleon ok I think we've decided on creating an `AnySendableHashable` type that ensures the `Hashable` types are `Sendable`. This should be fine for most keys unless people are doing something silly

Reply:
I think I have all of the (non-example code) Sendable warning wrangled now except for this one
```text
console-kit/Sources/ConsoleKit/Activity/ActivityIndicator.swift:80:41: warning: type 'any DispatchSourceTimer' does not conform to the 'Sendable' protocol
        self._timer = NIOLockedValueBox(DispatchSource.makeTimerSource(flags: [], queue: self.queue))
                                        ^
Dispatch.DispatchSourceTimer:1:17: note: protocol 'DispatchSourceTimer' does not conform to the 'Sendable' protocol
public protocol DispatchSourceTimer : DispatchSourceProtocol {
```

I would guess that basically all of the types in libdispatch should probably be Sendable, so I'm not sure if this is an oversight on the part of the library or a larger issue.

Reply:
You should be able get around this by writing it this way:
```swift
        self._timer = NIOLockedValueBox(DispatchSource.makeTimerSource(flags: [], queue: self.queue) as DispatchSource)
```
(you may need `as!` instead of `as`)

The trick is that while the `DispatchSource` class itself is `Sendable`, neither `DispatchSourceProtocol` nor the various subprotocols which represent the various specific source types in Swift (such as `DispatchSourceTimer`) follow suit. I believe this is an oversight in the API.